### PR TITLE
Create a development env using docker and docker compose

### PR DESCRIPTION
**Pull request by dmitryrck**
I think we can create a develpoment env using Docker and [Docker Compose](https://docs.docker.com/compose/).

I followed these files:
- `Dockerfile`, of couse ;)
- `doc/QUICK_START.md` and;
- `doc/development/setting-up-development-environment.md`;

### CODE REVIEW

**Comment by TeatroIO** [comment]
I've prepared a stage to preview changes. <a href='http://dmitryrck-docker-dev.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com/' target='_blank'>Open stage</a> or <a href='http://dmitryrck-docker-dev.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com//teatro.log' target='_blank'>view logs</a>.

**Comment by dmitryrck** [comment]
I'm sorry I missed to include `[ci skip]` in the commit :disappointed:

**Comment by oliverguenther** [comment]
Hey @dmitryrck, lots of thanks for your contribution!

This looks good to me, I'd like to give this a whirl and also give @crohr a chance to look on it (which will probably happen ~thursday).

Best,
Oliver

**Comment by crohr** [comment]
@dmitryrck thanks for that! One question though: any reason you didn't use the original Dockerfile? It's supposed to be working in dev mode as well.

File: docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+worker: &ruby
+  build: .
+  dockerfile: Dockerfile.dev
+  volumes:
+    - .:/app
+  links:
+    - db
+  command: "foreman start 'worker=1'"
+
+web-dev:
```
Comment (crohr) [inline docker-compose.yml]: Why not just `web`?

File: doc/QUICK_START.md [hunk 2]
```diff
@@ -34,10 +34,44 @@ Detailed installation instructions for different platforms are located on the [O
...
 
+## Fast install (Docker version)
+
+### Prerequisites
+
+* Git
+* [Docker](http://www.docker.com/)
+* [Docker Compose](https://docs.docker.com/compose/)
+
+To install Docker you can follow the [Get Started with Docker](https://docs.docker.com/mac/) guide.
```
Comment (crohr) [inline doc/QUICK_START.md]: I think you should reference https://docs.docker.com/engine/installation/ instead.

**Comment by crohr** [comment]
@dmitryrck I created #4273 as an alternative to reuse the existing Dockerfile, and have a process to automatically update assets. Let me know what you think!

File: docker-compose.yml (hunk 1 above)
Comment (dmitryrck) [inline docker-compose.yml]: Maybe in a near future `web` could be only for production ;)

But I changed it…

File: doc/QUICK_START.md (hunk 2 above)
Comment (dmitryrck) [inline doc/QUICK_START.md]: :+1:

**Comment by dmitryrck** [comment]
I think that we can simplify the development version, not using another user neither automatizing others, for example…

**Comment by crohr** [comment]
@dmitryrck I'm going to merge #4273 for now, mainly because I'm not too fond of having to maintain multiple Dockerfiles and different ways of launching the processes. However I want to thank you again for your work, which helped for the other PR. I also reverted to using a database YAML file instead of env variable, to make it easier to launch tests.

**Comment by dmitryrck** [comment]
:+1:

**Pull request closed without merging**